### PR TITLE
Force read resume_epoch as int & save actually used mean & std in model config.json

### LINKED ISSUE #55: Epoch number is treated as float when overwriting config from command line
image
Also tried "24" and '24'. None of them worked.

### PR DESCRIPTION
**Pull request by waynehuu**
Did two things:
1. Save mean & std that's actually used for transformation/normalization in model's config.json, rather than saving the default values stored in training config file (which most likely will be overridden in training). 
2. fix #55

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -93,6 +93,7 @@ def train_model(args, device, parallel):
     # make data loader
     ds_cfgs = [a for a in sorted(args.keys()) if 'dataset' in a]
```
Comment (waynehuu) [inline train.py:94]: Not so sure what this line does. Are there more than one 'dataset' keys expected?

File: train.py [hunk 2]
```diff
@@ -93,6 +93,7 @@ def train_model(args, device, parallel):
...
     assert ds_cfgs[0] == 'dataset'
+    mean, std = args[ds_cfgs[0]]['mean'], args[ds_cfgs[0]]['std'] # read default mean and std first
```
Comment (waynehuu) [inline train.py:96]: Added these to bound mean and std which are referenced in line 140. Previously they only appear in a loop as local variables (line 105).

File: train.py [hunk 3]
```diff
@@ -119,6 +122,7 @@ def train_model(args, device, parallel):
                 batch_size=int(args[ds_cfg]['batch_size']), shuffle=False, num_workers=int(args[ds_cfg]['num_workers']))
             print('Training model on the {} dataset'.format(args[ds_cfg]['ds_name']))
             train_val_loaders['valid'].append(valid_loader)
+    misc_utils.save_file(os.path.join(args['save_dir'], 'config.json'), args) # save config with actual mean and std used
```
Comment (waynehuu) [inline train.py:125]: Moved config.json saving from main() to train_model() as the values actually used in training will be saved.

File: train.py (hunk 1 above)
Comment (bohaohuang) [reply train.py:94]: This is for mixed batch ratio experiments where you can specify the batch size for multiple datasets

**Pull request closed without merging**